test/integration/binary/binaries_test_fixture.hpp

```diff
@@ -44,86 +44,143 @@ namespace shared_model {
 
 namespace query_validation {
 
+  using QueryIterator = std::vector<proto::Query>::iterator;
+
   namespace internal {
 
+    /**
+     * Helper type that is used as a marker of the end of types list in variadic
+     * templates.
+     * Should not be used outside of query_validation namespace.
+     */
     class Void {};
 
-  }  // namespace internal
-
-  using QueryIterator = std::vector<proto::Query>::iterator;
-
-  template <typename ResponseType>
-  inline void checkQueryResponse(
-      const shared_model::proto::QueryResponse &response) {
-    ASSERT_NO_THROW(boost::apply_visitor(
-        framework::SpecifiedVisitor<ResponseType>(), response.get()));
-  }
+    /**
+     * Asserts that actual query response type equals to expected type.
+     *
+     * @tparam ExpectedResponseType - expected type of QueryResponse object
+     * @param response - QueryResponse object to check
+     */
+    template <typename ExpectedResponseType>
+    inline void checkQueryResponseType(
```

Comment:
Would be better to return bool, instead of assertion
Otherwise fails will look like `Assertion failed at binaries_test_fixture.hpp:67`, that is pretty useless